ccp/src/prover.rs

```diff
@@ -111,8 +117,8 @@ impl NoxCCPApi for CCProver {
 }
 
 impl ToCCStatus for CCProver {
-    fn status(&self) -> CCStatus {
-        self.status
+    async fn status(&self) -> CCStatus {
```

Comment:
This looks rather dangerous.

`status().await` looks super innocent, but underline is Mutex. It is supper easy to write a deadlock this way.

Suggest to move all the locks above the API level. Also, it might make sense to change to RWLock if you need to read it independently from write.

Reply:
This particular piece of code seems to be unused and can be removed.

Also, RwLock is slower that Mutex, and if reads are short, Mutex can be more effective.